server/team/team.route.js

```diff
@@ -5,9 +5,20 @@ const { ifNoUserRedirect } = require('../middlewares/user');
 const { ifNoBody400 } = require('../middlewares/util');
 const teamCtrl = require('./team.contoller');
 const validators = require('./team.validation');
+const User = require('../user/user.model');
 
 const router = express.Router(); // eslint-disable-line new-cap
 
+function deleteValidator(req, res, next) {
+  return User.findOne({ _id: req.user._id })
```

Comment:
We don't need to hit the database here. Use `req.user.teamId` instead. But you will also need to check for a falsey `teamId`

I was thinking something like 

```js
function ifNotOwnTeam400(redirect = '/') {
  return function (req, res, next) {
    const { teamId: usersTeam } = req.user;
    const { teamId } = req.params;
    if (!usersTeam || usersTeam !== teamId) {
      return res.sendStatus(400).end();
    }
    return next();
  };
}
```